File: SportWatch/src/main/java/com/reactive/SportWatch/models/StreamingInfo.java
package com.reactive.SportWatch.models;

import java.net.URL;
import java.time.LocalDateTime;

// Info thats going to be sent to the frontend.
// streamUrl can be playbackUrl or presignedUrl of an uploaded stream so .m3u8 or .mp4
// thumbnailUrl always is presigned url
public record StreamingInfo(String title, String category,
                            URL streamUrl, URL thumbnailUrl,
                            int authorId, String desc,
                            LocalDateTime createdAt) {}

File: SportWatch_frontend/src/app/shared/interfaces/StreamingInfo.ts
export interface StreamingInfo {
  author: string;
  authorId: number;
  streamId: number;
  isLive: boolean;
  title: string;
  category: string;
  streamUrl: string;     // Using string for URL
  thumbnailUrl: string;  // Using string for URL
  desc: string;
  createdAt: string;     // ISO date string (LocalDateTime -> string)
  viewerCount?: number; // Property set on the frontend.
}
